src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_SIZE = 10;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index do not exist");
+        }
+        if (array.length == size) {
+            grow();
+        }
+        fastValueAdd(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexExistCheck(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexExistCheck(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexExistCheck(index);
+        final T result = array[index];
+        fastRemove(index);
+        return result;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        final T result;
+        final int newSize = size - 1;
+
+        for (int i = 0; i <= newSize; i++) {
```

Comment:
you don't need an additional variable here. Don't add the redundant line
```suggestion
        for (int i = 0; i < size; i++) {
```